template/listEntry/listEntry.proto

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package listEntry;
+
+import "pkg/adapter/template/TemplateExtensions.proto";
+
+option (istio.mixer.v1.config.template.template_variety) = TEMPLATE_VARIETY_CHECK;
+option (istio.mixer.v1.config.template.template_name) = "ListEntry";
+
+// ListEntry is used to verify the presence/absence of a string
+// within a list.
+message Template {
+    // Specifies the entry to verify in the list.
+    string value = 1;
```

Comment:
Supported types are like a white list we don't support all types in the template. Main reason for supporting finite set is we want to create this magic Instance object and the generated code that wire up the things inside mixer. The generated code is data type specific (for example if it is a map<x,y> when we generated code in go we want to do a `for x, y := range fooMap, for non map we tackle different for list we will have to write the for loop slightly differently). It is definitely doable to extend the vocab of supported types in Template message if we have a use case. We just started with a small set.

The currently supported types are:
string, double, float, bool, ValueType and map<string, ValueType>

I am currently working on supporting few more types 
protobuf.datetime, protobuf.duration.